sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowPythonRunner.scala

```diff
@@ -96,6 +101,8 @@ class ArrowPythonRunner(
             writer.writeBatch()
             arrowWriter.reset()
           }
+          val deltaData = dataOut.size() - startData
+          pythonDataSent += deltaData
```

Comment:
If you intend to measure data output size during writing to python worker, should you move the two lines into the loop so it can keep tracking output size after each batch?